realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -765,6 +783,28 @@ public Builder compactOnLaunch(CompactOnLaunchCallback compactOnLaunch) {
             return this;
         }
 
+        /**
+         * Sets the maximum number of live versions in the Realm file before an {@link IllegalStateException} is thrown when
+         * attempting to write more data.
+         * <p>
+         * Realm is capable of concurrently handling many different versions of Realm objects. This can e.g. happen if you
+         * have a Realm open on many different threads or are freezing objects while data is being written to the file.
+         * <p>
+         * Under normal circumstances this is not a problem, but if the number of active versions grow too large, it will
+         * have a negative effect on the filesize on disk. Setting this parameters can therefore be used to prevent uses of
+         * Realm that can result in very large Realms.
+         *
+         * @param number the maximum number of active versions before an exception is thrown.
+         * @see <a href="https://realm.io/docs/java/latest/#faq-large-realm-file-size">FAQ</a>
+         */
+        public Builder maxNumberOfActiveVersions(long number) {
```

Comment:
Yes, the number includes versions produced by Sync. I agree it can get a bit misleading in that case. The problem is that the number of versions is really just a bit poor proxy for what we really care about. Namely that the file doesn't grow out of control.

IMO, this is still useful, but I will update the docs to be more clear about what it is.